src/wasm/literal.cpp

```diff
@@ -2604,6 +2604,21 @@ Literal Literal::dotSI16x8toI32x4(const Literal& other) const {
   return dot<4, 2, &Literal::getLanesSI16x8>(*this, other);
 }
 
+Literal Literal::dotSI8x16toI16x8Add(const Literal& left,
+                                     const Literal& right) const {
+  auto temp = dotSI8x16toI16x8(left);
+
+  auto tempLanes = temp.getLanesSI16x8();
+  LaneArray<4> dest;
+  // TODO: the index on dest may be wrong, see
+  //       https://github.com/WebAssembly/relaxed-simd/issues/162
```

Comment:
I guess we don't need the TODO any more if we know that the upstream tests pass.

Reply:
Yeah, and the fuzzer seems fine, so I'm guessing this is correct... but maybe I'll wait for confirmation from the people there before removing this.